src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -267,16 +255,11 @@ app.directive('dashboardApp', function ($injector) {
 
       $scope.$watch(() => dashboardState.getAppState().$newFilters, function (filters = []) {
         // need to convert filters generated from user interaction with viz into kuery AST
-        // normally these would be handled by the filter bar directive
-        if ($scope.model.query.language === 'kuery') {
-          const kueryAST = fromKueryExpression($scope.model.query.query);
-
-          const newKueryAST = filters.reduce((currentAST, filter) => {
-            return addNode(currentAST, filterToKueryAST(filter));
-          }, kueryAST);
-
-          $scope.updateQuery({ query: toKueryExpression(newKueryAST), language: 'kuery' });
-        }
+        // These are handled by the filter bar directive when lucene is the query language
+        Promise.all(filters.map(queryManager.addLegacyFilter))
+        .then(updateState)
```

Comment:
I've updated this spot to maintain consistency with the rest of this particular file

30029ebfdedaf1da6baa9fc240637d0bcc62d9b0